From e36798dff45660ed7e07b15be09621791f826f41 Mon Sep 17 00:00:00 2001
From: chenguangli <chenguangli2@huawei.com>
Date: Sat, 4 Sep 2021 17:22:49 +0800
Subject: [PATCH] scsi/hifc: Fix memory leakage bug
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: e36798dff45660ed7e07b15be09621791f826f41


driver inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I47Z9P

-----------------------------------------------------------------------

The memory will be leaked when driver set the port_poto_cfg invalid value.

Signed-off-by: chenguangli <chenguangli2@huawei.com>
Reviewed-by: baowenyi <baowenyi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/scsi/huawei/hifc/hifc_chipitf.c | 2 +-
 drivers/scsi/huawei/hifc/unf_common.h   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/scsi/huawei/hifc/hifc_chipitf.c b/drivers/scsi/huawei/hifc/hifc_chipitf.c
index fe97f6468557..74e3df2fe860 100644
--- a/drivers/scsi/huawei/hifc/hifc_chipitf.c
+++ b/drivers/scsi/huawei/hifc/hifc_chipitf.c
@@ -369,7 +369,7 @@ unsigned int hifc_config_port_table(struct hifc_hba_s *v_hba)
 			   v_hba->port_cfg.port_id,
 			   (unsigned char)v_hba->port_topo_cfg);
 
-		return UNF_RETURN_ERROR;
+		goto exit;
 	}
 
 	/* About speed */
diff --git a/drivers/scsi/huawei/hifc/unf_common.h b/drivers/scsi/huawei/hifc/unf_common.h
index bc11cc542c56..9253cc3eada5 100644
--- a/drivers/scsi/huawei/hifc/unf_common.h
+++ b/drivers/scsi/huawei/hifc/unf_common.h
@@ -13,7 +13,7 @@
 /* B version, B0XX Corresponding x.x */
 #define UNF_B_VERSION "5.0"
 /* Indicates the minor version number of the driver */
-#define UNF_DRIVER_VERSION "11"
+#define UNF_DRIVER_VERSION "12"
 /* version num */
 #define UNF_FC_VERSION UNF_MAJOR_VERSION "." UNF_B_VERSION "." UNF_DRIVER_VERSION
 extern unsigned int unf_dbg_level;
-- 
2.26.2

